R/period_dataset.R

```diff
@@ -0,0 +1,449 @@
+#'   period variables. It is not allowed to mix for example period and subperiod
+#'   variables. If period *and* subperiod variables are required, separate calls
+#'   must be used.
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of quosures where the expressions are symbols as returned by
+#'   `vars()` is expected.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @details For each subperiod/period/phase in the period reference dataset and
+#'   each element in `new_vars` a variable (LHS value of `new_vars`) is added to
+#'   the output dataset and set to the value of the source variable (RHS value
+#'   of `new_vars`.
+#'
+#' @return The input dataset with subperiod/period/phase variables added (see
+#'   "Details" section)
+#'
+#' @seealso [create_period_dataset()]
+#'
+#' @keywords der_adsl
+#' @family der_adsl
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#' library(lubridate)
+#'
+#' adsl <- tibble(STUDYID = "xyz", USUBJID = c("1", "2"))
+#'
+#' # Add period variables to ADSL
+#' period_ref <- tribble(
+#'   ~USUBJID, ~APERIOD, ~APERSDT,     ~APEREDT,
+#'   "1",             1, "2021-01-04", "2021-02-06",
+#'   "1",             2, "2021-02-07", "2021-03-07",
+#'   "2",             1, "2021-02-02", "2021-03-02",
+#'   "2",             2, "2021-03-03", "2021-04-01"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "xyz",
+#'     APERIOD = as.integer(APERIOD),
+#'     across(matches("APER[ES]DT"), ymd)
+#'   )
+#'
+#' derive_vars_period(
+#'   adsl,
+#'   dataset_ref = period_ref,
+#'   new_vars = vars(APxxSDT = APERSDT, APxxEDT = APEREDT)
+#' ) %>%
+#'   select(STUDYID, USUBJID, AP01SDT, AP01EDT, AP02SDT, AP02EDT)
+#'
+#' # Add phase variables to ADSL
+#' phase_ref <- tribble(
+#'   ~USUBJID, ~APHASEN, ~PHSDT,       ~PHEDT,       ~APHASE,
+#'   "1",             1, "2021-01-04", "2021-02-06", "TREATMENT",
+#'   "1",             2, "2021-02-07", "2021-03-07", "FUP",
+#'   "2",             1, "2021-02-02", "2021-03-02", "TREATMENT"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "xyz",
+#'     APHASEN = as.integer(APHASEN),
+#'     across(matches("PH[ES]DT"), ymd)
+#'   )
+#'
+#' derive_vars_period(
+#'   adsl,
+#'   dataset_ref = phase_ref,
+#'   new_vars = vars(PHwSDT = PHSDT, PHwEDT = PHEDT, APHASEw = APHASE)
+#' ) %>%
+#'   select(STUDYID, USUBJID, PH1SDT, PH1EDT, PH2SDT, PH2EDT, APHASE1, APHASE2)
+#'
+#' # Add subperiod variables to ADSL
+#' subperiod_ref <- tribble(
+#'   ~USUBJID, ~APERIOD, ~ASPER, ~ASPRSDT,     ~ASPREDT,
+#'   "1",             1,      1, "2021-01-04", "2021-01-19",
+#'   "1",             1,      2, "2021-01-20", "2021-02-06",
+#'   "1",             2,      1, "2021-02-07", "2021-03-07",
+#'   "2",             1,      1, "2021-02-02", "2021-03-02",
+#'   "2",             2,      1, "2021-03-03", "2021-04-01"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "xyz",
+#'     APERIOD = as.integer(APERIOD),
+#'     ASPER = as.integer(ASPER),
+#'     across(matches("APER[ES]DT"), ymd)
```

Comment:
![image](https://user-images.githubusercontent.com/43018762/197214194-b7ea4ecb-efe5-47bf-b65f-35a5731eb812.png)

Saw the date columns rendering as chr, this should address it

```suggestion
#'     across(matches("ASPR[ES]DT"), ymd)
```